### PR TITLE
PACT-1003: refactor to avoid decoding twice

### PR DESCRIPTION
**Pull request by rwalpole**
In this PR the `BusinessServiceRequest` trait has been removed and replaced with a `RequestMessage` trait in a new package `uk.gov.nationalarchives.omega.api.messages.request`. There is also a corresponding package for reply messages with a `ReplyMessage` trait. All incoming messages, as described in the AsyncAPI schema, are now case classes extending the `RequestMessage` trait and all outgoing message are case classes extending the `ReplyMessage` trait. The `BusinessRequestValidation` trait `validateRequest` function now returns a `ValidationResult` of type `RequestMessage`. This means that the `BusinessService` trait's `process` function now takes a `RequestMessage` parameter which can then be easily converted to the expected type using `request.asInstanceOf[..]`. Although we should check for a `ClassCastException` at this point, there is no realistic prospect of this occuring as the correct type will have already been decoded in the new `decodePayload` function of the `Dispatcher`. The `decodePayload` and `dispatchRequest` functions within the `Dispatcher` now handle the decoding of the payload, service selection and execution in place of the previous `createAndValidateServiceRequest`, `createServiceRequest`, `validateBusinessServiceRequest` and `execBusinessService` functions, thereby simplifying the code significantly.

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/omega/api/messages/request/ListAssetLegalStatusSummary.scala [hunk 1]
```diff
@@ -19,8 +19,8 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-package uk.gov.nationalarchives.omega.api.business.legalstatus
+package uk.gov.nationalarchives.omega.api.messages.request
 
 import uk.gov.nationalarchives.omega.api.business.BusinessServiceRequest
```
Comment (userjd) [inline src/main/scala/uk/gov/nationalarchives/omega/api/messages/request/ListAssetLegalStatusSummary.scala:25]: unused import

File: src/main/scala/uk/gov/nationalarchives/omega/api/business/BusinessService.scala [hunk 2]
```diff
@@ -21,6 +21,8 @@
 
 package uk.gov.nationalarchives.omega.api.business
 
+import uk.gov.nationalarchives.omega.api.messages.request.RequestMessage
+
 trait BusinessService {
-  def process(request: BusinessServiceRequest): Either[BusinessServiceError, BusinessServiceReply]
+  def process(request: RequestMessage): Either[BusinessServiceError, BusinessServiceReply]
```
Comment (userjd) [inline src/main/scala/uk/gov/nationalarchives/omega/api/business/BusinessService.scala:27]: could `BusinessServiceRequest` be deleted

File: src/main/scala/uk/gov/nationalarchives/omega/api/business/BusinessRequestValidation.scala [hunk 3]
```diff
@@ -22,10 +22,11 @@
 package uk.gov.nationalarchives.omega.api.business
 
 import cats.data.ValidatedNec
```
Comment (userjd) [inline src/main/scala/uk/gov/nationalarchives/omega/api/business/BusinessRequestValidation.scala:24]: unused import

File: src/test/scala/uk/gov/nationalarchives/omega/api/business/agents/ListAgentServiceSummarySpec.scala [hunk 4]
```diff
@@ -55,7 +52,7 @@ class ListAgentServiceSummarySpec extends AnyFreeSpec with Matchers {
 
       "an empty payload request" in {
 
-        val listAgentSummaryRequest = ListAgentSummaryRequest()
+        val listAgentSummaryRequest = ListAgentSummary(List())
```
Comment (userjd) [inline src/test/scala/uk/gov/nationalarchives/omega/api/business/agents/ListAgentServiceSummarySpec.scala:55]: For this scenario the stub data is being returned however, after connecting to the database what is the expected results w/ or w/o the depository

**Comment by userjd** [comment]
Suggest to rename `AgentSummaryError` be renamed to `ListAgentSummaryError` for consistency even though there is no change for this in the PR

File: src/main/scala/uk/gov/nationalarchives/omega/api/messages/request/ListAssetLegalStatusSummary.scala (hunk 1 above)
Comment (rwalpole) [reply src/main/scala/uk/gov/nationalarchives/omega/api/messages/request/ListAssetLegalStatusSummary.scala]: Removed.

File: src/main/scala/uk/gov/nationalarchives/omega/api/business/BusinessService.scala (hunk 2 above)
Comment (rwalpole) [reply src/main/scala/uk/gov/nationalarchives/omega/api/business/BusinessService.scala:27]: Removed.

File: src/main/scala/uk/gov/nationalarchives/omega/api/business/BusinessRequestValidation.scala (hunk 3 above)
Comment (rwalpole) [reply src/main/scala/uk/gov/nationalarchives/omega/api/business/BusinessRequestValidation.scala]: Removed.

File: src/test/scala/uk/gov/nationalarchives/omega/api/business/agents/ListAgentServiceSummarySpec.scala (hunk 4 above)
Comment (rwalpole) [reply src/test/scala/uk/gov/nationalarchives/omega/api/business/agents/ListAgentServiceSummarySpec.scala:57]: Good question. I have set this test to be ignored until PACT-1025 is resolved.

**Review by userjd — APPROVED** [review]

**Pull request merged**